src/core/probe/kernel/bpf/include/common.h

```diff
@@ -286,11 +295,13 @@ static __always_inline int chain(struct retis_context *ctx)
 	CALL_HOOK(8)
 	CALL_HOOK(9)
 
-	if (get_event_size(event) > pass_threshold)
+	if (get_event_size(event) > pass_threshold) {
 		send_event(event);
-	else
+	} else {
 discard_event:
+		err_report(ctx->ksym, 0);
```

Comment:
> This doesn't work because there are valid cases to jump here:
> 
>     * Hook filtered the event.

while giving a careful look at it, I noticed something that I missed in a previous review here.
Unfortunately, we cannot rely on -ENOMSG (I guess in skb based modules) because attaching the program to a tracepoint with hooks returning non-zero value results in:

```
At program exit the register R0 has value (0xffffffd6; 0x0) should have been in (0x0; 0x0)
processed 901 insns (limit 1000000) max_states_per_insn 0 total_states 47 peak_states 47 mark_read 41
-- END PROG LOAD LOG --
libbpf: prog 'hook': failed to load: -22
libbpf: failed to load object 'hook'
Error: System error, errno: 22

```

Filtering worked around it. I think we should file an issue for this.

Reply:
given that we know sections will succeed and are not counted for hooks, and given that we can't rely on hooks retval (because of the previous comment), I think the best thing to do is to narrow the scope of the counters to its actual and initial purpose (ringbuf exhaustion) and report to user-space if get_event() failed. Other types of error will be added (event sections errors, if they can happen). Furthermore some might be better suited for some kind of dedicated trace pipe rather than counters and may benefit more from that kind of error reporting facility.

Reply:
I would turn dropped_events into ringbuf_errors and change the warn!().
`{} ring buffer error(s) in {}`

Reply:
ack, works for me